// @adyen/api-library/lib/src/services/balancePlaftform/balanceAccounts.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const models_1 = require("../../typings/balancePlatform/models");
const balancePlaftformResource_1 = __importDefault(require("../resource/balancePlaftformResource"));
class BalanceAccounts extends service_1.default {
    async create(request) {
        const resource = new balancePlaftformResource_1.default(this, "/balanceAccounts");
        const response = await (0, getJsonResponse_1.default)(resource, request, { method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "BalanceAccount");
    }
    async listSweeps(balanceAccountId, requestOptions) {
        const resource = new balancePlaftformResource_1.default(this, `/balanceAccounts/${balanceAccountId}/sweeps`);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "BalanceSweepConfigurationsResponse");
    }
    async createSweep(balanceAccountId, request) {
        const resource = new balancePlaftformResource_1.default(this, `/balanceAccounts/${balanceAccountId}/sweeps`);
        const response = await (0, getJsonResponse_1.default)(resource, request, { method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "SweepConfigurationV2");
    }
    async deleteSweep(balanceAccountId, sweepId) {
        const resource = new balancePlaftformResource_1.default(this, `/balanceAccounts/${balanceAccountId}/sweeps/${sweepId}`);
        await (0, getJsonResponse_1.default)(resource, "", { method: "DELETE" });
    }
    async retrieveSweep(balanceAccountId, sweepId) {
        const resource = new balancePlaftformResource_1.default(this, `/balanceAccounts/${balanceAccountId}/sweeps/${sweepId}`);
        const response = await (0, getJsonResponse_1.default)(resource, "", { method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "SweepConfigurationV2");
    }
    async updateSweep(balanceAccountId, sweepId, request) {
        const resource = new balancePlaftformResource_1.default(this, `/balanceAccounts/${balanceAccountId}/sweeps/${sweepId}`);
        const response = await (0, getJsonResponse_1.default)(resource, request, { method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "SweepConfigurationV2");
    }
    async retrieve(id) {
        const resource = new balancePlaftformResource_1.default(this, `/balanceAccounts/${id}`);
        const response = await (0, getJsonResponse_1.default)(resource, "", { method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "BalanceAccount");
    }
    async update(id, request) {
        const resource = new balancePlaftformResource_1.default(this, `/balanceAccounts/${id}`);
        const response = await (0, getJsonResponse_1.default)(resource, request, { method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "BalanceAccount");
    }
    async listPaymentInstruments(id, requestOptions) {
        const resource = new balancePlaftformResource_1.default(this, `/balanceAccounts/${id}/paymentInstruments`);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "PaginatedPaymentInstrumentsResponse");
    }
}
exports.default = BalanceAccounts;
