// @adyen/api-library/lib/src/services/balancePlatform.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const service_1 = __importDefault(require("../service"));
const accountHolders_1 = __importDefault(require("./balancePlaftform/accountHolders"));
const balanceAccounts_1 = __importDefault(require("./balancePlaftform/balanceAccounts"));
const general_1 = __importDefault(require("./balancePlaftform/general"));
const paymentInstruments_1 = __importDefault(require("./balancePlaftform/paymentInstruments"));
const paymentInstrumentGroups_1 = __importDefault(require("./balancePlaftform/paymentInstrumentGroups"));
const transactionRules_1 = __importDefault(require("./balancePlaftform/transactionRules"));
class BalancePlatform extends service_1.default {
    constructor(client) {
        super(client);
    }
    get General() {
        return new general_1.default(this.client);
    }
    get AccountHolders() {
        return new accountHolders_1.default(this.client);
    }
    get BalanceAccounts() {
        return new balanceAccounts_1.default(this.client);
    }
    get PaymentInstruments() {
        return new paymentInstruments_1.default(this.client);
    }
    get PaymentInstrumentGroups() {
        return new paymentInstrumentGroups_1.default(this.client);
    }
    get TransactionRules() {
        return new transactionRules_1.default(this.client);
    }
}
exports.default = BalancePlatform;
