// @adyen/api-library/lib/src/services/legalEntityManagement.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const service_1 = __importDefault(require("../service"));
const legalEntities_1 = __importDefault(require("./legalEntityManagement/legalEntities"));
const hostedOnboardingPage_1 = __importDefault(require("./legalEntityManagement/hostedOnboardingPage"));
const transferInstruments_1 = __importDefault(require("./legalEntityManagement/transferInstruments"));
const businessLineService_1 = __importDefault(require("./legalEntityManagement/businessLineService"));
const documents_1 = __importDefault(require("./legalEntityManagement/documents"));
class LegalEntityManagement extends service_1.default {
    constructor(client) {
        super(client);
    }
    get LegalEntities() {
        return new legalEntities_1.default(this.client);
    }
    get TransferInstruments() {
        return new transferInstruments_1.default(this.client);
    }
    get BusinessLineService() {
        return new businessLineService_1.default(this.client);
    }
    get Documents() {
        return new documents_1.default(this.client);
    }
    get HostedOnboardingPage() {
        return new hostedOnboardingPage_1.default(this.client);
    }
}
exports.default = LegalEntityManagement;
