// @adyen/api-library/lib/src/services/recurring.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const getJsonResponse_1 = __importDefault(require("../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../service"));
const models_1 = require("../typings/recurring/models");
const RecurringResource_1 = __importDefault(require("./resource/RecurringResource"));
class Recurring extends service_1.default {
    constructor(client) {
        super(client);
        this._listRecurringDetails = new RecurringResource_1.default(this, "/listRecurringDetails");
        this._disable = new RecurringResource_1.default(this, "/disable");
        this._scheduleAccountUpdater = new RecurringResource_1.default(this, "/scheduleAccountUpdater");
        this._notifyShopper = new RecurringResource_1.default(this, "/notifyShopper");
        this._createPermit = new RecurringResource_1.default(this, "/createPermit");
        this._disablePermit = new RecurringResource_1.default(this, "/disablePermit");
    }
    listRecurringDetails(recurringDetailsRequest) {
        const request = models_1.ObjectSerializer.serialize(recurringDetailsRequest, "RecurringDetailsRequest");
        return (0, getJsonResponse_1.default)(this._listRecurringDetails, request);
    }
    disable(disableRequest) {
        const request = models_1.ObjectSerializer.serialize(disableRequest, "DisableRequest");
        return (0, getJsonResponse_1.default)(this._disable, request);
    }
    scheduleAccountUpdater(scheduleAccountUpdaterRequest) {
        const request = models_1.ObjectSerializer.serialize(scheduleAccountUpdaterRequest, "ScheduleAccountUpdaterRequest");
        return (0, getJsonResponse_1.default)(this._scheduleAccountUpdater, request);
    }
    notifyShopper(notifyShopperRequest) {
        const request = models_1.ObjectSerializer.serialize(notifyShopperRequest, "NotifyShopperRequest");
        return (0, getJsonResponse_1.default)(this._notifyShopper, request);
    }
    createPermit(createPermitRequest) {
        const request = models_1.ObjectSerializer.serialize(createPermitRequest, "CreatePermitRequest");
        return (0, getJsonResponse_1.default)(this._createPermit, request);
    }
    disablePermit(disablePermitRequest) {
        const request = models_1.ObjectSerializer.serialize(disablePermitRequest, "DisablePermitRequest");
        return (0, getJsonResponse_1.default)(this._disablePermit, request);
    }
}
exports.default = Recurring;
