// @adyen/api-library/lib/src/services/storedValue.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const getJsonResponse_1 = __importDefault(require("../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../service"));
const changeStatus_1 = __importDefault(require("./resource/storedValue/changeStatus"));
const issue_1 = __importDefault(require("./resource/storedValue/issue"));
const load_1 = __importDefault(require("./resource/storedValue/load"));
const checkBalance_1 = __importDefault(require("./resource/storedValue/checkBalance"));
const mergeBalance_1 = __importDefault(require("./resource/storedValue/mergeBalance"));
const voidTransaction_1 = __importDefault(require("./resource/storedValue/voidTransaction"));
class StoredValue extends service_1.default {
    constructor(client) {
        super(client);
        this._issue = new issue_1.default(this);
        this._changeStatus = new changeStatus_1.default(this);
        this._load = new load_1.default(this);
        this._checkBalance = new checkBalance_1.default(this);
        this._mergebalance = new mergeBalance_1.default(this);
        this._voidTransaction = new voidTransaction_1.default(this);
    }
    issue(request) {
        return (0, getJsonResponse_1.default)(this._issue, request);
    }
    changeStatus(request) {
        return (0, getJsonResponse_1.default)(this._changeStatus, request);
    }
    load(request) {
        return (0, getJsonResponse_1.default)(this._load, request);
    }
    checkBalance(request) {
        return (0, getJsonResponse_1.default)(this._checkBalance, request);
    }
    mergebalance(request) {
        return (0, getJsonResponse_1.default)(this._mergebalance, request);
    }
    voidTransaction(request) {
        return (0, getJsonResponse_1.default)(this._voidTransaction, request);
    }
}
exports.default = StoredValue;
