// @adyen/api-library/lib/src/services/terminalManagement.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const getJsonResponse_1 = __importDefault(require("../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../service"));
const models_1 = require("../typings/terminalManagement/models");
const assignTerminals_1 = __importDefault(require("./resource/terminalManagement/assignTerminals"));
const findTerminal_1 = __importDefault(require("./resource/terminalManagement/findTerminal"));
const getStoresUnderAccount_1 = __importDefault(require("./resource/terminalManagement/getStoresUnderAccount"));
const getTerminalDetails_1 = __importDefault(require("./resource/terminalManagement/getTerminalDetails"));
const getTerminalsUnderAccount_1 = __importDefault(require("./resource/terminalManagement/getTerminalsUnderAccount"));
class TerminalManagement extends service_1.default {
    constructor(client) {
        super(client);
        this._assignTerminals = new assignTerminals_1.default(this);
        this._findTerminal = new findTerminal_1.default(this);
        this._getStoresUnderAccount = new getStoresUnderAccount_1.default(this);
        this._getTerminalDetails = new getTerminalDetails_1.default(this);
        this._getTerminalsUnderAccount = new getTerminalsUnderAccount_1.default(this);
    }
    async assignTerminals(request) {
        const response = await (0, getJsonResponse_1.default)(this._assignTerminals, request);
        return models_1.ObjectSerializer.deserialize(response, "AssignTerminalsResponse");
    }
    async findTerminal(request) {
        const response = await (0, getJsonResponse_1.default)(this._findTerminal, request);
        return models_1.ObjectSerializer.deserialize(response, "FindTerminalResponse");
    }
    async getStoresUnderAccount(request) {
        const response = await (0, getJsonResponse_1.default)(this._getStoresUnderAccount, request);
        return models_1.ObjectSerializer.deserialize(response, "GetStoresUnderAccountResponse");
    }
    async getTerminalDetails(request) {
        const response = await (0, getJsonResponse_1.default)(this._getTerminalDetails, request);
        return models_1.ObjectSerializer.deserialize(response, "GetTerminalDetailsResponse");
    }
    async getTerminalsUnderAccount(request) {
        const response = await (0, getJsonResponse_1.default)(this._getTerminalsUnderAccount, request);
        return models_1.ObjectSerializer.deserialize(response, "GetTerminalsUnderAccountResponse");
    }
}
exports.default = TerminalManagement;
