// @adyen/api-library/lib/src/client.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const config_1 = __importDefault(require("./config"));
const httpURLConnectionClient_1 = __importDefault(require("./httpClient/httpURLConnectionClient"));
const package_json_1 = require("../package.json");
class Client {
    constructor(options) {
        if (options.config) {
            this.config = options.config;
        }
        else {
            this.config = new config_1.default();
        }
        const environment = options.environment || this.config.environment;
        if (environment) {
            this.setEnvironment(environment, options.liveEndpointUrlPrefix);
            if (options.username && options.password) {
                this.config.username = options.username;
                this.config.password = options.password;
                if (options.applicationName) {
                    this.config.applicationName = options.applicationName;
                }
            }
            if (options.apiKey) {
                this.config.apiKey = options.apiKey;
            }
        }
        if (options.httpClient) {
            this._httpClient = options.httpClient;
        }
    }
    setEnvironment(environment, liveEndpointUrlPrefix) {
        if (environment === "TEST") {
            this.config.endpoint = Client.ENDPOINT_TEST;
            this.config.marketPayEndpoint = Client.MARKETPAY_ENDPOINT_TEST;
            this.config.hppEndpoint = Client.HPP_TEST;
            this.config.checkoutEndpoint = Client.CHECKOUT_ENDPOINT_TEST;
            this.config.terminalApiCloudEndpoint = Client.TERMINAL_API_ENDPOINT_TEST;
            this.config.paymentEndpoint = Client.PAYMENT_API_ENDPOINT_TEST;
            this.config.storedValueEndpoint = Client.STOREDVALUE_API_ENDPOINT_TEST;
            this.config.terminalManagementEndpoint = Client.TERMINAL_MANAGEMENT_API_ENDPOINT_TEST;
            this.config.managementEndpoint = Client.MANAGEMENT_API_ENDPOINT_TEST;
            this.config.balancePlatformEndpoint = Client.BALANCE_PLATFORM_API_ENDPOINT_TEST;
            this.config.legalEntityManagementEndpoint = Client.LEGAL_ENTITY_MANAGEMENT_API_ENDPOINT_TEST;
            this.config.transfersEndpoint = Client.TRANSFERS_API_ENDPOINT_TEST;
            this.config.dataProtectionEndpoint = Client.DATA_PROTECTION_API_ENDPOINT_TEST;
        }
        else if (environment === "LIVE") {
            this.config.endpoint = Client.ENDPOINT_LIVE;
            this.config.marketPayEndpoint = Client.MARKETPAY_ENDPOINT_LIVE;
            this.config.hppEndpoint = Client.HPP_LIVE;
            this.config.terminalApiCloudEndpoint = Client.TERMINAL_API_ENDPOINT_LIVE;
            this.config.paymentEndpoint = Client.PAYMENT_API_ENDPOINT_LIVE;
            this.config.storedValueEndpoint = Client.STOREDVALUE_API_ENDPOINT_LIVE;
            this.config.terminalManagementEndpoint = Client.TERMINAL_MANAGEMENT_API_ENDPOINT_LIVE;
            this.config.managementEndpoint = Client.MANAGEMENT_API_ENDPOINT_LIVE;
            this.config.balancePlatformEndpoint = Client.BALANCE_PLATFORM_API_ENDPOINT_LIVE;
            this.config.legalEntityManagementEndpoint = Client.LEGAL_ENTITY_MANAGEMENT_API_ENDPOINT_LIVE;
            this.config.transfersEndpoint = Client.TRANSFERS_API_ENDPOINT_LIVE;
            this.config.dataProtectionEndpoint = Client.DATA_PROTECTION_API_ENDPOINT_LIVE;
            if (liveEndpointUrlPrefix) {
                this.config.endpoint =
                    `${Client.ENDPOINT_PROTOCOL}${liveEndpointUrlPrefix}${Client.ENDPOINT_LIVE_SUFFIX}`;
                this.config.checkoutEndpoint =
                    `${Client.ENDPOINT_PROTOCOL}${liveEndpointUrlPrefix}${Client.CHECKOUT_ENDPOINT_LIVE_SUFFIX}`;
            }
            else {
                this.config.endpoint = Client.ENDPOINT_LIVE;
                this.config.checkoutEndpoint = undefined;
            }
        }
    }
    get httpClient() {
        if (!this._httpClient) {
            this._httpClient = new httpURLConnectionClient_1.default();
        }
        return this._httpClient;
    }
    set httpClient(httpClient) {
        this._httpClient = httpClient;
    }
    setApplicationName(applicationName) {
        this.config.applicationName = applicationName;
    }
    setTimeouts(connectionTimeoutMillis, readTimeoutMillis) {
        this.config.connectionTimeoutMillis = connectionTimeoutMillis;
        this.config.readTimeoutMillis = readTimeoutMillis;
    }
}
Client.ENDPOINT_TEST = "https://pal-test.adyen.com";
Client.ENDPOINT_LIVE = "https://pal-live.adyen.com";
Client.ENDPOINT_LIVE_SUFFIX = "-pal-live.adyenpayments.com";
Client.HPP_TEST = "https://test.adyen.com/hpp";
Client.HPP_LIVE = "https://live.adyen.com/hpp";
Client.MARKETPAY_ENDPOINT_TEST = "https://cal-test.adyen.com/cal/services";
Client.MARKETPAY_ENDPOINT_LIVE = "https://cal-live.adyen.com/cal/services";
Client.CHECKOUT_API_VERSION = "v70";
Client.API_VERSION = "v68";
Client.RECURRING_API_VERSION = "v68";
Client.MARKETPAY_ACCOUNT_API_VERSION = "v6";
Client.MARKETPAY_FUND_API_VERSION = "v6";
Client.MARKETPAY_HOP_API_VERSION = "v6";
Client.MARKETPAY_NOTIFICATION_API_VERSION = "v5";
Client.MARKETPAY_NOTIFICATION_CONFIGURATION_API_VERSION = "v6";
Client.PAYMENT_API_VERSION = "v68";
Client.STOREDVALUE_API_VERSION = "v46";
Client.TERMINAL_MANAGEMENT_API_VERSION = "v1";
Client.MANAGEMENT_API_VERSION = "v1";
Client.LIB_NAME = "adyen-node-api-library";
Client.LIB_VERSION = package_json_1.version;
Client.CHECKOUT_ENDPOINT_TEST = "https://checkout-test.adyen.com/checkout";
Client.CHECKOUT_ENDPOINT_LIVE_SUFFIX = "-checkout-live.adyenpayments.com/checkout";
Client.BIN_LOOKUP_PAL_SUFFIX = "/pal/servlet/BinLookup/";
Client.BIN_LOOKUP_API_VERSION = "v50";
Client.TERMINAL_API_ENDPOINT_TEST = "https://terminal-api-test.adyen.com";
Client.TERMINAL_API_ENDPOINT_LIVE = "https://terminal-api-live.adyen.com";
Client.ENDPOINT_PROTOCOL = "https://";
Client.PAYMENT_API_ENDPOINT_TEST = "https://pal-test.adyen.com/pal/servlet/Payment";
Client.PAYMENT_API_ENDPOINT_LIVE = "https://pal-live.adyen.com/pal/servlet/Payment";
Client.STOREDVALUE_API_ENDPOINT_TEST = "https://pal-test.adyen.com/pal/servlet/StoredValue";
Client.STOREDVALUE_API_ENDPOINT_LIVE = "https://pal-live.adyen.com/pal/servlet/StoredValue";
Client.TERMINAL_MANAGEMENT_API_ENDPOINT_TEST = "https://postfmapi-test.adyen.com/postfmapi/terminal";
Client.TERMINAL_MANAGEMENT_API_ENDPOINT_LIVE = "https://postfmapi-live.adyen.com/postfmapi/terminal";
Client.MANAGEMENT_API_ENDPOINT_TEST = "https://management-test.adyen.com";
Client.MANAGEMENT_API_ENDPOINT_LIVE = "https://management-live.adyen.com";
Client.BALANCE_PLATFORM_API_VERSION = "v2";
Client.BALANCE_PLATFORM_API_ENDPOINT_TEST = "https://balanceplatform-api-test.adyen.com/bcl";
Client.BALANCE_PLATFORM_API_ENDPOINT_LIVE = "https://balanceplatform-api-live.adyen.com/bcl";
Client.LEGAL_ENTITY_MANAGEMENT_API_VERSION = "v2";
Client.LEGAL_ENTITY_MANAGEMENT_API_ENDPOINT_TEST = "https://kyc-test.adyen.com/lem";
Client.LEGAL_ENTITY_MANAGEMENT_API_ENDPOINT_LIVE = "https://kyc-live.adyen.com/lem";
Client.TRANSFERS_API_VERSION = "v3";
Client.TRANSFERS_API_ENDPOINT_TEST = "https://balanceplatform-api-test.adyen.com/btl";
Client.TRANSFERS_API_ENDPOINT_LIVE = "https://balanceplatform-api-live.adyen.com/btl";
Client.DATA_PROTECTION_API_VERSION = "v1";
Client.DATA_PROTECTION_API_ENDPOINT_TEST = "https://ca-test.adyen.com/ca/services/DataProtectionService";
Client.DATA_PROTECTION_API_ENDPOINT_LIVE = "https://ca-live.adyen.com/ca/services/DataProtectionService";
exports.default = Client;
