// @adyen/api-library/lib/src/typings/checkout/checkoutBalanceCheckResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutBalanceCheckResponse = void 0;
class CheckoutBalanceCheckResponse {
    static getAttributeTypeMap() {
        return CheckoutBalanceCheckResponse.attributeTypeMap;
    }
}
exports.CheckoutBalanceCheckResponse = CheckoutBalanceCheckResponse;
CheckoutBalanceCheckResponse.discriminator = undefined;
CheckoutBalanceCheckResponse.attributeTypeMap = [
    {
        "name": "additionalData",
        "baseName": "additionalData",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "balance",
        "baseName": "balance",
        "type": "Amount"
    },
    {
        "name": "fraudResult",
        "baseName": "fraudResult",
        "type": "FraudResult"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "refusalReason",
        "baseName": "refusalReason",
        "type": "string"
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "CheckoutBalanceCheckResponse.ResultCodeEnum"
    },
    {
        "name": "transactionLimit",
        "baseName": "transactionLimit",
        "type": "Amount"
    }
];
(function (CheckoutBalanceCheckResponse) {
    let ResultCodeEnum;
    (function (ResultCodeEnum) {
        ResultCodeEnum["Success"] = "Success";
        ResultCodeEnum["NotEnoughBalance"] = "NotEnoughBalance";
        ResultCodeEnum["Failed"] = "Failed";
    })(ResultCodeEnum = CheckoutBalanceCheckResponse.ResultCodeEnum || (CheckoutBalanceCheckResponse.ResultCodeEnum = {}));
})(CheckoutBalanceCheckResponse = exports.CheckoutBalanceCheckResponse || (exports.CheckoutBalanceCheckResponse = {}));
