// @adyen/api-library/lib/src/typings/checkout/createPaymentRefundRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreatePaymentRefundRequest = void 0;
class CreatePaymentRefundRequest {
    static getAttributeTypeMap() {
        return CreatePaymentRefundRequest.attributeTypeMap;
    }
}
exports.CreatePaymentRefundRequest = CreatePaymentRefundRequest;
CreatePaymentRefundRequest.discriminator = undefined;
CreatePaymentRefundRequest.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "lineItems",
        "baseName": "lineItems",
        "type": "Array<LineItem>"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "merchantRefundReason",
        "baseName": "merchantRefundReason",
        "type": "CreatePaymentRefundRequest.MerchantRefundReasonEnum"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "splits",
        "baseName": "splits",
        "type": "Array<Split>"
    }
];
(function (CreatePaymentRefundRequest) {
    let MerchantRefundReasonEnum;
    (function (MerchantRefundReasonEnum) {
        MerchantRefundReasonEnum["Fraud"] = "FRAUD";
        MerchantRefundReasonEnum["CustomerRequest"] = "CUSTOMER REQUEST";
        MerchantRefundReasonEnum["Return"] = "RETURN";
        MerchantRefundReasonEnum["Duplicate"] = "DUPLICATE";
        MerchantRefundReasonEnum["Other"] = "OTHER";
    })(MerchantRefundReasonEnum = CreatePaymentRefundRequest.MerchantRefundReasonEnum || (CreatePaymentRefundRequest.MerchantRefundReasonEnum = {}));
})(CreatePaymentRefundRequest = exports.CreatePaymentRefundRequest || (exports.CreatePaymentRefundRequest = {}));
