// @adyen/api-library/lib/src/typings/checkout/createPaymentAmountUpdateRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreatePaymentAmountUpdateRequest = void 0;
class CreatePaymentAmountUpdateRequest {
    static getAttributeTypeMap() {
        return CreatePaymentAmountUpdateRequest.attributeTypeMap;
    }
}
exports.CreatePaymentAmountUpdateRequest = CreatePaymentAmountUpdateRequest;
CreatePaymentAmountUpdateRequest.discriminator = undefined;
CreatePaymentAmountUpdateRequest.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "industryUsage",
        "baseName": "industryUsage",
        "type": "CreatePaymentAmountUpdateRequest.IndustryUsageEnum"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "splits",
        "baseName": "splits",
        "type": "Array<Split>"
    }
];
(function (CreatePaymentAmountUpdateRequest) {
    let IndustryUsageEnum;
    (function (IndustryUsageEnum) {
        IndustryUsageEnum["DelayedCharge"] = "delayedCharge";
        IndustryUsageEnum["Installment"] = "installment";
        IndustryUsageEnum["NoShow"] = "noShow";
    })(IndustryUsageEnum = CreatePaymentAmountUpdateRequest.IndustryUsageEnum || (CreatePaymentAmountUpdateRequest.IndustryUsageEnum = {}));
})(CreatePaymentAmountUpdateRequest = exports.CreatePaymentAmountUpdateRequest || (exports.CreatePaymentAmountUpdateRequest = {}));
