// @adyen/api-library/lib/src/typings/checkout/donationResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DonationResponse = void 0;
class DonationResponse {
    static getAttributeTypeMap() {
        return DonationResponse.attributeTypeMap;
    }
}
exports.DonationResponse = DonationResponse;
DonationResponse.discriminator = undefined;
DonationResponse.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "donationAccount",
        "baseName": "donationAccount",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "payment",
        "baseName": "payment",
        "type": "PaymentResponse"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "DonationResponse.StatusEnum"
    }
];
(function (DonationResponse) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Completed"] = "completed";
        StatusEnum["Pending"] = "pending";
        StatusEnum["Refused"] = "refused";
    })(StatusEnum = DonationResponse.StatusEnum || (DonationResponse.StatusEnum = {}));
})(DonationResponse = exports.DonationResponse || (exports.DonationResponse = {}));
