// @adyen/api-library/lib/src/typings/checkout/paymentAmountUpdateResource.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentAmountUpdateResource = void 0;
class PaymentAmountUpdateResource {
    static getAttributeTypeMap() {
        return PaymentAmountUpdateResource.attributeTypeMap;
    }
}
exports.PaymentAmountUpdateResource = PaymentAmountUpdateResource;
PaymentAmountUpdateResource.discriminator = undefined;
PaymentAmountUpdateResource.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "industryUsage",
        "baseName": "industryUsage",
        "type": "PaymentAmountUpdateResource.IndustryUsageEnum"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "paymentPspReference",
        "baseName": "paymentPspReference",
        "type": "string"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "splits",
        "baseName": "splits",
        "type": "Array<Split>"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "PaymentAmountUpdateResource.StatusEnum"
    }
];
(function (PaymentAmountUpdateResource) {
    let IndustryUsageEnum;
    (function (IndustryUsageEnum) {
        IndustryUsageEnum["DelayedCharge"] = "delayedCharge";
        IndustryUsageEnum["Installment"] = "installment";
        IndustryUsageEnum["NoShow"] = "noShow";
    })(IndustryUsageEnum = PaymentAmountUpdateResource.IndustryUsageEnum || (PaymentAmountUpdateResource.IndustryUsageEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Received"] = "received";
    })(StatusEnum = PaymentAmountUpdateResource.StatusEnum || (PaymentAmountUpdateResource.StatusEnum = {}));
})(PaymentAmountUpdateResource = exports.PaymentAmountUpdateResource || (exports.PaymentAmountUpdateResource = {}));
