// @adyen/api-library/lib/src/typings/checkout/paymentRefundResource.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentRefundResource = void 0;
class PaymentRefundResource {
    static getAttributeTypeMap() {
        return PaymentRefundResource.attributeTypeMap;
    }
}
exports.PaymentRefundResource = PaymentRefundResource;
PaymentRefundResource.discriminator = undefined;
PaymentRefundResource.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "lineItems",
        "baseName": "lineItems",
        "type": "Array<LineItem>"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "merchantRefundReason",
        "baseName": "merchantRefundReason",
        "type": "PaymentRefundResource.MerchantRefundReasonEnum"
    },
    {
        "name": "paymentPspReference",
        "baseName": "paymentPspReference",
        "type": "string"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "splits",
        "baseName": "splits",
        "type": "Array<Split>"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "PaymentRefundResource.StatusEnum"
    }
];
(function (PaymentRefundResource) {
    let MerchantRefundReasonEnum;
    (function (MerchantRefundReasonEnum) {
        MerchantRefundReasonEnum["Fraud"] = "FRAUD";
        MerchantRefundReasonEnum["CustomerRequest"] = "CUSTOMER REQUEST";
        MerchantRefundReasonEnum["Return"] = "RETURN";
        MerchantRefundReasonEnum["Duplicate"] = "DUPLICATE";
        MerchantRefundReasonEnum["Other"] = "OTHER";
    })(MerchantRefundReasonEnum = PaymentRefundResource.MerchantRefundReasonEnum || (PaymentRefundResource.MerchantRefundReasonEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Received"] = "received";
    })(StatusEnum = PaymentRefundResource.StatusEnum || (PaymentRefundResource.StatusEnum = {}));
})(PaymentRefundResource = exports.PaymentRefundResource || (exports.PaymentRefundResource = {}));
