// @adyen/api-library/lib/src/typings/management/shopperStatement.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShopperStatement = void 0;
class ShopperStatement {
    static getAttributeTypeMap() {
        return ShopperStatement.attributeTypeMap;
    }
}
exports.ShopperStatement = ShopperStatement;
ShopperStatement.discriminator = undefined;
ShopperStatement.attributeTypeMap = [
    {
        "name": "doingBusinessAsName",
        "baseName": "doingBusinessAsName",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "ShopperStatement.TypeEnum"
    }
];
(function (ShopperStatement) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Append"] = "append";
        TypeEnum["Dynamic"] = "dynamic";
        TypeEnum["Fixed"] = "fixed";
    })(TypeEnum = ShopperStatement.TypeEnum || (ShopperStatement.TypeEnum = {}));
})(ShopperStatement = exports.ShopperStatement || (exports.ShopperStatement = {}));
