// @adyen/api-library/lib/src/services/management/accountStoreLevelApi.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const managementResource_1 = __importDefault(require("../resource/management/managementResource"));
const models_1 = require("../../typings/management/models");
class AccountStoreLevelApi extends service_1.default {
    /**
     * @summary Get a list of stores
     * @param merchantId The unique identifier of the merchant account.
     * @param pageNumber The number of the page to fetch.
     * @param pageSize The number of items to have on a page, maximum 100. The default is 10 items on a page.
     * @param reference The reference of the store.
     */
    async listStoresByMerchantId(merchantId, requestOptions) {
        const localVarPath = "/merchants/{merchantId}/stores"
            .replace('{' + 'merchantId' + '}', encodeURIComponent(String(merchantId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "ListStoresResponse");
    }
    /**
     * @summary Get a store
     * @param merchantId The unique identifier of the merchant account.
     * @param storeId The unique identifier of the store.
     */
    async getStore(merchantId, storeId, requestOptions) {
        const localVarPath = "/merchants/{merchantId}/stores/{storeId}"
            .replace('{' + 'merchantId' + '}', encodeURIComponent(String(merchantId)))
            .replace('{' + 'storeId' + '}', encodeURIComponent(String(storeId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "Store");
    }
    /**
     * @summary Get a list of stores
     * @param pageNumber The number of the page to fetch.
     * @param pageSize The number of items to have on a page, maximum 100. The default is 10 items on a page.
     * @param reference The reference of the store.
     * @param merchantId The unique identifier of the merchant account.
     */
    async listStores(requestOptions) {
        const localVarPath = "/stores";
        const resource = new managementResource_1.default(this, localVarPath);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "ListStoresResponse");
    }
    /**
     * @summary Get a store
     * @param storeId The unique identifier of the store.
     */
    async getStoreById(storeId, requestOptions) {
        const localVarPath = "/stores/{storeId}"
            .replace('{' + 'storeId' + '}', encodeURIComponent(String(storeId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "Store");
    }
    /**
     * @summary Update a store
     * @param merchantId The unique identifier of the merchant account.
     * @param storeId The unique identifier of the store.
     * @param updateStoreRequest
     */
    async updateStore(merchantId, storeId, updateStoreRequest, requestOptions) {
        const localVarPath = "/merchants/{merchantId}/stores/{storeId}"
            .replace('{' + 'merchantId' + '}', encodeURIComponent(String(merchantId)))
            .replace('{' + 'storeId' + '}', encodeURIComponent(String(storeId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const request = models_1.ObjectSerializer.serialize(updateStoreRequest, "UpdateStoreRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "Store");
    }
    /**
     * @summary Update a store
     * @param storeId The unique identifier of the store.
     * @param updateStoreRequest
     */
    async updateStoreById(storeId, updateStoreRequest, requestOptions) {
        const localVarPath = "/stores/{storeId}"
            .replace('{' + 'storeId' + '}', encodeURIComponent(String(storeId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const request = models_1.ObjectSerializer.serialize(updateStoreRequest, "UpdateStoreRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "Store");
    }
    /**
     * @summary Create a store
     * @param merchantId The unique identifier of the merchant account.
     * @param storeCreationRequest
     */
    async createStoreByMerchantId(merchantId, storeCreationRequest, requestOptions) {
        const localVarPath = "/merchants/{merchantId}/stores"
            .replace('{' + 'merchantId' + '}', encodeURIComponent(String(merchantId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const request = models_1.ObjectSerializer.serialize(storeCreationRequest, "StoreCreationRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "Store");
    }
    /**
     * @summary Create a store
     * @param storeCreationWithMerchantCodeRequest
     */
    async createStore(storeCreationWithMerchantCodeRequest, requestOptions) {
        const localVarPath = "/stores";
        const resource = new managementResource_1.default(this, localVarPath);
        const request = models_1.ObjectSerializer.serialize(storeCreationWithMerchantCodeRequest, "StoreCreationWithMerchantCodeRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "Store");
    }
}
exports.default = AccountStoreLevelApi;
