// @adyen/api-library/lib/src/services/management/accountMerchantLevelApi.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const managementResource_1 = __importDefault(require("../resource/management/managementResource"));
const models_1 = require("../../typings/management/models");
class AccountMerchantLevelApi extends service_1.default {
    /**
     * @summary Get a list of merchant accounts
     * @param pageNumber The number of the page to fetch.
     * @param pageSize The number of items to have on a page, maximum 100. The default is 10 items on a page.
     */
    async listMerchantAccounts(requestOptions) {
        const localVarPath = "/merchants";
        const resource = new managementResource_1.default(this, localVarPath);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "ListMerchantResponse");
    }
    /**
     * @summary Get a merchant account
     * @param merchantId The unique identifier of the merchant account.
     */
    async getMerchantAccount(merchantId, requestOptions) {
        const localVarPath = "/merchants/{merchantId}"
            .replace('{' + 'merchantId' + '}', encodeURIComponent(String(merchantId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "Merchant");
    }
    /**
     * @summary Create a merchant account
     * @param createMerchantRequest
     */
    async createMerchantAccount(createMerchantRequest, requestOptions) {
        const localVarPath = "/merchants";
        const resource = new managementResource_1.default(this, localVarPath);
        const request = models_1.ObjectSerializer.serialize(createMerchantRequest, "CreateMerchantRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "CreateMerchantResponse");
    }
    /**
     * @summary Request to activate a merchant account
     * @param merchantId The unique identifier of the merchant account.
     */
    async requestToActivateMerchantAccount(merchantId, requestOptions) {
        const localVarPath = "/merchants/{merchantId}/activate"
            .replace('{' + 'merchantId' + '}', encodeURIComponent(String(merchantId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "RequestActivationResponse");
    }
}
exports.default = AccountMerchantLevelApi;
