// @adyen/api-library/lib/src/services/management/merchantPayoutSettings.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const models_1 = require("../../typings/management/models");
const managementResource_1 = __importDefault(require("../resource/management/managementResource"));
class MerchantPayoutSettings extends service_1.default {
    /**
     * Get a list of payout settings
     */
    async listPayoutSettings(merchantId) {
        const resource = new managementResource_1.default(this, `/merchants/${merchantId}/payoutSettings`);
        const response = await (0, getJsonResponse_1.default)(resource, "", { method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "PayoutSettingsResponse");
    }
    /**
     * Add a payout setting
     */
    async create(merchantId, request) {
        const resource = new managementResource_1.default(this, `/merchants/${merchantId}/payoutSettings`);
        const response = await (0, getJsonResponse_1.default)(resource, request, { method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "PayoutSettings");
    }
    /**
     * Delete a payout setting
     */
    async delete(merchantId, payoutSettingsId) {
        const resource = new managementResource_1.default(this, `/merchants/${merchantId}/payoutSettings/${payoutSettingsId}`);
        await (0, getJsonResponse_1.default)(resource, "", { method: "DELETE" });
    }
    /**
     * Get a payout setting
     */
    async retrieve(merchantId, payoutSettingsId) {
        const resource = new managementResource_1.default(this, `/merchants/${merchantId}/payoutSettings/${payoutSettingsId}`);
        const response = await (0, getJsonResponse_1.default)(resource, "", { method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "PayoutSettings");
    }
    /**
     * Update a payout setting
     */
    async update(merchantId, payoutSettingsId, request) {
        const resource = new managementResource_1.default(this, `/merchants/${merchantId}/payoutSettings/${payoutSettingsId}`);
        const response = await (0, getJsonResponse_1.default)(resource, request, { method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "PayoutSettings");
    }
}
exports.default = MerchantPayoutSettings;
