// @adyen/api-library/lib/src/services/management/paymentMethodsMerchantLevelApi.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const managementResource_1 = __importDefault(require("../resource/management/managementResource"));
const models_1 = require("../../typings/management/models");
class PaymentMethodsMerchantLevelApi extends service_1.default {
    /**
     * @summary Get all payment methods
     * @param merchantId The unique identifier of the merchant account.
     * @param storeId The unique identifier of the store for which to return the payment methods.
     * @param businessLineId The unique identifier of the Business Line for which to return the payment methods.
     * @param pageSize The number of items to have on a page, maximum 100. The default is 10 items on a page.
     * @param pageNumber The number of the page to fetch.
     */
    async getAllPaymentMethods(merchantId, requestOptions) {
        const localVarPath = "/merchants/{merchantId}/paymentMethodSettings"
            .replace('{' + 'merchantId' + '}', encodeURIComponent(String(merchantId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "PaymentMethodResponse");
    }
    /**
     * @summary Get payment method details
     * @param merchantId The unique identifier of the merchant account.
     * @param paymentMethodId The unique identifier of the payment method.
     */
    async getPaymentMethodDetails(merchantId, paymentMethodId, requestOptions) {
        const localVarPath = "/merchants/{merchantId}/paymentMethodSettings/{paymentMethodId}"
            .replace('{' + 'merchantId' + '}', encodeURIComponent(String(merchantId)))
            .replace('{' + 'paymentMethodId' + '}', encodeURIComponent(String(paymentMethodId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "PaymentMethod");
    }
    /**
     * @summary Get Apple Pay domains
     * @param merchantId The unique identifier of the merchant account.
     * @param paymentMethodId The unique identifier of the payment method.
     */
    async getApplePayDomains(merchantId, paymentMethodId, requestOptions) {
        const localVarPath = "/merchants/{merchantId}/paymentMethodSettings/{paymentMethodId}/getApplePayDomains"
            .replace('{' + 'merchantId' + '}', encodeURIComponent(String(merchantId)))
            .replace('{' + 'paymentMethodId' + '}', encodeURIComponent(String(paymentMethodId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "ApplePayInfo");
    }
    /**
     * @summary Update a payment method
     * @param merchantId The unique identifier of the merchant account.
     * @param paymentMethodId The unique identifier of the payment method.
     * @param updatePaymentMethodInfo
     */
    async updatePaymentMethod(merchantId, paymentMethodId, updatePaymentMethodInfo, requestOptions) {
        const localVarPath = "/merchants/{merchantId}/paymentMethodSettings/{paymentMethodId}"
            .replace('{' + 'merchantId' + '}', encodeURIComponent(String(merchantId)))
            .replace('{' + 'paymentMethodId' + '}', encodeURIComponent(String(paymentMethodId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const request = models_1.ObjectSerializer.serialize(updatePaymentMethodInfo, "UpdatePaymentMethodInfo");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "PaymentMethod");
    }
    /**
     * @summary Request a payment method
     * @param merchantId The unique identifier of the merchant account.
     * @param paymentMethodSetupInfo
     */
    async requestPaymentMethod(merchantId, paymentMethodSetupInfo, requestOptions) {
        const localVarPath = "/merchants/{merchantId}/paymentMethodSettings"
            .replace('{' + 'merchantId' + '}', encodeURIComponent(String(merchantId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const request = models_1.ObjectSerializer.serialize(paymentMethodSetupInfo, "PaymentMethodSetupInfo");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "PaymentMethod");
    }
    /**
     * @summary Add an Apple Pay domain
     * @param merchantId The unique identifier of the merchant account.
     * @param paymentMethodId The unique identifier of the payment method.
     * @param applePayInfo
     */
    async addApplePayDomain(merchantId, paymentMethodId, applePayInfo, requestOptions) {
        const localVarPath = "/merchants/{merchantId}/paymentMethodSettings/{paymentMethodId}/addApplePayDomains"
            .replace('{' + 'merchantId' + '}', encodeURIComponent(String(merchantId)))
            .replace('{' + 'paymentMethodId' + '}', encodeURIComponent(String(paymentMethodId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const request = models_1.ObjectSerializer.serialize(applePayInfo, "ApplePayInfo");
        await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
    }
}
exports.default = PaymentMethodsMerchantLevelApi;
