// @adyen/api-library/lib/src/services/management/terminalSettingsTerminalLevelApi.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const managementResource_1 = __importDefault(require("../resource/management/managementResource"));
const models_1 = require("../../typings/management/models");
class TerminalSettingsTerminalLevelApi extends service_1.default {
    /**
     * @summary Get the terminal logo
     * @param terminalId The unique identifier of the payment terminal.
     */
    async getTerminalLogo(terminalId, requestOptions) {
        const localVarPath = "/terminals/{terminalId}/terminalLogos"
            .replace('{' + 'terminalId' + '}', encodeURIComponent(String(terminalId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "Logo");
    }
    /**
     * @summary Get terminal settings
     * @param terminalId The unique identifier of the payment terminal.
     */
    async getTerminalSettings(terminalId, requestOptions) {
        const localVarPath = "/terminals/{terminalId}/terminalSettings"
            .replace('{' + 'terminalId' + '}', encodeURIComponent(String(terminalId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "TerminalSettings");
    }
    /**
     * @summary Update the logo
     * @param terminalId The unique identifier of the payment terminal.
     * @param logo
     */
    async updateLogo(terminalId, logo, requestOptions) {
        const localVarPath = "/terminals/{terminalId}/terminalLogos"
            .replace('{' + 'terminalId' + '}', encodeURIComponent(String(terminalId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const request = models_1.ObjectSerializer.serialize(logo, "Logo");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "Logo");
    }
    /**
     * @summary Update terminal settings
     * @param terminalId The unique identifier of the payment terminal.
     * @param terminalSettings
     */
    async updateTerminalSettings(terminalId, terminalSettings, requestOptions) {
        const localVarPath = "/terminals/{terminalId}/terminalSettings"
            .replace('{' + 'terminalId' + '}', encodeURIComponent(String(terminalId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const request = models_1.ObjectSerializer.serialize(terminalSettings, "TerminalSettings");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "TerminalSettings");
    }
}
exports.default = TerminalSettingsTerminalLevelApi;
