// @adyen/api-library/lib/src/services/management/usersMerchantLevelApi.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const managementResource_1 = __importDefault(require("../resource/management/managementResource"));
const models_1 = require("../../typings/management/models");
class UsersMerchantLevelApi extends service_1.default {
    /**
     * @summary Get a list of users
     * @param merchantId Unique identifier of the merchant.
     * @param pageNumber The number of the page to fetch.
     * @param pageSize The number of items to have on a page. Maximum value is **100**. The default is **10** items on a page.
     */
    async listUsers(merchantId, requestOptions) {
        const localVarPath = "/merchants/{merchantId}/users"
            .replace('{' + 'merchantId' + '}', encodeURIComponent(String(merchantId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "ListMerchantUsersResponse");
    }
    /**
     * @summary Get user details
     * @param merchantId Unique identifier of the merchant.
     * @param userId Unique identifier of the user.
     */
    async getUserDetails(merchantId, userId, requestOptions) {
        const localVarPath = "/merchants/{merchantId}/users/{userId}"
            .replace('{' + 'merchantId' + '}', encodeURIComponent(String(merchantId)))
            .replace('{' + 'userId' + '}', encodeURIComponent(String(userId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "User");
    }
    /**
     * @summary Update a user
     * @param merchantId Unique identifier of the merchant.
     * @param userId Unique identifier of the user.
     * @param updateMerchantUserRequest
     */
    async updateUser(merchantId, userId, updateMerchantUserRequest, requestOptions) {
        const localVarPath = "/merchants/{merchantId}/users/{userId}"
            .replace('{' + 'merchantId' + '}', encodeURIComponent(String(merchantId)))
            .replace('{' + 'userId' + '}', encodeURIComponent(String(userId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const request = models_1.ObjectSerializer.serialize(updateMerchantUserRequest, "UpdateMerchantUserRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "User");
    }
    /**
     * @summary Create a new user
     * @param merchantId Unique identifier of the merchant.
     * @param createMerchantUserRequest
     */
    async createNewUser(merchantId, createMerchantUserRequest, requestOptions) {
        const localVarPath = "/merchants/{merchantId}/users"
            .replace('{' + 'merchantId' + '}', encodeURIComponent(String(merchantId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const request = models_1.ObjectSerializer.serialize(createMerchantUserRequest, "CreateMerchantUserRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "CreateUserResponse");
    }
}
exports.default = UsersMerchantLevelApi;
