// @adyen/api-library/lib/src/services/management/usersCompanyLevelApi.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const managementResource_1 = __importDefault(require("../resource/management/managementResource"));
const models_1 = require("../../typings/management/models");
class UsersCompanyLevelApi extends service_1.default {
    /**
     * @summary Get a list of users
     * @param companyId The unique identifier of the company account.
     * @param pageNumber The number of the page to return.
     * @param pageSize The number of items to have on a page. Maximum value is **100**. The default is **10** items on a page.
     */
    async listUsers(companyId, requestOptions) {
        const localVarPath = "/companies/{companyId}/users"
            .replace('{' + 'companyId' + '}', encodeURIComponent(String(companyId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "ListCompanyUsersResponse");
    }
    /**
     * @summary Get user details
     * @param companyId The unique identifier of the company account.
     * @param userId The unique identifier of the user.
     */
    async getUserDetails(companyId, userId, requestOptions) {
        const localVarPath = "/companies/{companyId}/users/{userId}"
            .replace('{' + 'companyId' + '}', encodeURIComponent(String(companyId)))
            .replace('{' + 'userId' + '}', encodeURIComponent(String(userId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "CompanyUser");
    }
    /**
     * @summary Update user details
     * @param companyId The unique identifier of the company account.
     * @param userId The unique identifier of the user.
     * @param updateCompanyUserRequest
     */
    async updateUserDetails(companyId, userId, updateCompanyUserRequest, requestOptions) {
        const localVarPath = "/companies/{companyId}/users/{userId}"
            .replace('{' + 'companyId' + '}', encodeURIComponent(String(companyId)))
            .replace('{' + 'userId' + '}', encodeURIComponent(String(userId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const request = models_1.ObjectSerializer.serialize(updateCompanyUserRequest, "UpdateCompanyUserRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "CompanyUser");
    }
    /**
     * @summary Create a new user
     * @param companyId The unique identifier of the company account.
     * @param createCompanyUserRequest
     */
    async createNewUser(companyId, createCompanyUserRequest, requestOptions) {
        const localVarPath = "/companies/{companyId}/users"
            .replace('{' + 'companyId' + '}', encodeURIComponent(String(companyId)));
        const resource = new managementResource_1.default(this, localVarPath);
        const request = models_1.ObjectSerializer.serialize(createCompanyUserRequest, "CreateCompanyUserRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "CreateCompanyUserResponse");
    }
}
exports.default = UsersCompanyLevelApi;
